cpp/ql/src/semmle/code/cpp/dataflow/EscapesTree.qll

```diff
@@ -212,14 +212,9 @@ private predicate addressMayEscapeAt(Expr e) {
 
 private predicate addressMayEscapeMutablyAt(Expr e) {
   addressMayEscapeAt(e) and
-  exists(Type t | t = e.getType().getUnderlyingType() |
-    exists(PointerType pt |
-      pt = t
-      or
-      pt = t.(SpecifiedType).getBaseType()
-    |
-      not pt.getBaseType().isConst()
-    )
+  exists(Type t | t = e.getType().getUnderlyingType().stripTopLevelSpecifiers() |
```

Comment:
```suggestion
  exists(Type t | t = e.getType().stripTopLevelSpecifiers() |
```

Does this work too?

Reply:
Yes, I convinced myself that this is indeed sufficient.

My reasoning:
The only difference whether we resolve *all* typedefs in that line or just stripping of the top-level typedef and specifiers is in the invocations of `getBaseType().isConst()`.
Indeed, `getBaseType()` now may return a different type (due to typedefs), however, `isConst()` resolves typedefs, per the documentation of `Type.hasSpecifier()`.